src/client/common/alias_dict.cpp

```diff
@@ -175,33 +175,31 @@ void mp::AliasDict::save_dict()
         aliases_json.insert(name, definition);
     }
 
-    QTemporaryFile temp_file;
+    auto config_file_name = QString::fromStdString(aliases_file);
+
+    QDir config_path = QFileInfo(config_file_name).absoluteDir();
+    if (!config_path.exists())
+        config_path.mkpath(config_path.absolutePath());
+
+    QTemporaryFile temp_file(config_file_name);
+
     if (temp_file.open())
     {
+        temp_file.setAutoRemove(false);
+
         auto temp_file_name = temp_file.fileName();
 
         mp::write_json(aliases_json, temp_file_name);
 
         temp_file.close();
 
-        auto config_file_name = QString::fromStdString(aliases_file);
-
         if (QFile::exists(config_file_name))
         {
             auto backup_file_name = config_file_name + ".bak";
             QFile::remove(backup_file_name);
-            QFile::copy(config_file_name, backup_file_name);
-            QFile::remove(config_file_name);
-        }
-        else
-        {
-            QDir config_path = QFileInfo(config_file_name).absoluteDir();
-            if (!config_path.exists())
-            {
-                config_path.mkpath(config_path.absolutePath());
-            }
+            QFile::rename(config_file_name, backup_file_name);
         }
 
-        QFile::copy(temp_file_name, config_file_name);
+        QFile::rename(temp_file_name, config_file_name);
     }
```

Comment:
Should probably handle errors here?

Reply:
Good idea. But... what to do if we can't rename? Error out?

In fact, I just fixed a bug in windows here. The `QTemporaryFile` refused to be renamed using `QFile::rename()`; instead, I had to use the non-static version of `rename`.

Reply:
Yeah, if renaming has an error, it should error out since the config file won't be consistent at this point.

Reply:
Right, done.